app/src/test/java/com/github/multimatum_team/multimatum/service/ProcrastinationDetectorServiceTest.kt

```diff
@@ -0,0 +1,249 @@
+    }
+
+    @After
+    fun release(){
+        Intents.release()
+    }
+
+    @Test
+    fun service_should_be_registered_as_listener_and_then_unregistered(){
+        val mockSensor: Sensor = mock()
+        var wasRegistered = false
+        var wasUnregistered = false
+        `when`(mockSensorManager.getDefaultSensor(eq(ProcrastinationDetectorService.REF_SENSOR))).thenReturn(mockSensor)
+        `when`(mockSensorManager.registerListener(isA<ProcrastinationDetectorService>(), eq(mockSensor), any())).then {
+            wasRegistered = true
+            true
+        }
+        `when`(mockSensorManager.unregisterListener(isA<ProcrastinationDetectorService>(), eq(mockSensor))).then {
+            assertThat("ProcrastinationDetectorService should be registered as a listener before being unregistered",
+                wasRegistered, `is`(true))
+            wasUnregistered = true
+            true
+        }
+        // create the controller; this calls onStartCommand on the service, which should call registerListener
+        controller.create().startCommand(0, 0)
+        assertThat("ProcrastinationDetectorService should be registered as a listener",
+            wasRegistered, `is`(true))
+        // the call to destroy calls onDestroy on the service, which should call unregisterListener
+        controller.destroy()
+        assertThat("ProcrastinationDetectorService should be unregistered",
+            wasUnregistered, `is`(true))
+    }
+
+    @Test
+    fun toast_should_be_displayed_when_sensor_detected_major_change() {
+        val mockSensor: Sensor = mock()
+        val mockSensorEvent: SensorEvent = mock()
+        `when`(mockSensorManager.getDefaultSensor(eq(ProcrastinationDetectorService.REF_SENSOR))).thenReturn(mockSensor)
+        val service = controller.create().startCommand(0, 0).get()
+        simulate2successiveToastEvents(
+            0f, 0f, 0f, 2_000_000_000,     // 1st event
+            10f, 10f, 10f, 5_000_000_000,  // 2nd event
+            mockSensorEvent, service
+        )
+        assertThatLastToastTextWas(applicationContext.getString(R.string.stop_procrastinating_msg))
+        controller.destroy()
+    }
+
+    @Test
+    fun toast_should_not_be_displayed_when_sensor_detected_tiny_change() {
+        val mockSensor: Sensor = mock()
+        val mockSensorEvent: SensorEvent = mock()
+        `when`(mockSensorManager.getDefaultSensor(eq(ProcrastinationDetectorService.REF_SENSOR))).thenReturn(mockSensor)
+        val service = controller.create().startCommand(0, 0).get()
+        simulate2successiveToastEvents(
+            0f, 0f, 0f, 2_000_000_000,                  // 1st event
+            TINY_CHANGE, TINY_CHANGE, TINY_CHANGE, 5_000_000_000,  // 2nd event
+            mockSensorEvent, service
+        )
+        assertThatLastToastTextWas(FAKE_TOAST_TEXT)
+        controller.destroy()
+    }
+
+    @Test
+    fun on_bind_should_return_a_binder_that_is_able_to_provide_the_bound_service(){
```

Comment:
Nitpick: you can use the ``fun `onBinds should return a binder...`() { ... }`` syntax. Oddly this syntax is not supported in androidTests but it does in regular unit tests.

Reply:
Actually I already had problems with this syntax where it appeared as `onBind%20should%20return%20a%20binder` (or similar) on the test results, so I prefer to keep it with underscores

Reply:
I see, then we'll probably have to refactor the other tests to use the normal syntax everywhere

Reply:
I don't think that it is worth the effort. I tried running all the tests and did not notice these %20 this time (actually when I noticed them it was in Intellij for another project, so the problem might not even exist on Android Studio, I just stick to underscore notation because I think that it is readable enough and avoid this kind of bugs in case they may arise somewhere). And it is no big deal if we use different conventions for the names of tests I would say.